### ISSUE TITLE
interval function not work

### ISSUE THREAD

**Issue by kainwen** [greenplum-db/gpdb#3023]
```
nyu=# select version();
                                                                                                          version
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 PostgreSQL 8.3.23 (Greenplum Database 5.0.0-beta.7+dev.109.g7c69d55 build dev) on x86_64-pc-linux-gnu, compiled by GCC gcc (GCC) 4.4.7 20120313 (Red Hat 4.4.7-18) compiled on Aug 22 2017 10:52:03 (with assert checking)
(1 row)

nyu=# select date '1999-11-11' + interval '1' day;
      ?column?
---------------------
 1999-11-11 00:00:00
(1 row)

nyu=# select date '1999-11-11' + interval '1 day';
      ?column?
---------------------
 1999-11-12 00:00:00
(1 row)
```

**Comment by hlinnaka** [comment]
I believe this is the same as issue #1187.

**Comment by davecramer** [comment]
Is the intent to leave this un-fixed in 5.0?

Dave Cramer

**Comment by hlinnaka** [comment]
On 08/23/2017 01:49 PM, Dave Cramer wrote:
> Is the intent to leave this un-fixed in 5.0?

Yeah, I believe so. In PostgreSQL, too, the decision back then was to 
not backport it. We're still in beta, so there's some room for arguing 
that we should backport it now, but I'd be happy to leave it until 6.0.

- Heikki

**Comment by davecramer** [comment]
On 23 August 2017 at 07:34, Heikki Linnakangas <notifications@github.com>
wrote:

> On 08/23/2017 01:49 PM, Dave Cramer wrote:
> > Is the intent to leave this un-fixed in 5.0?
>
> Yeah, I believe so. In PostgreSQL, too, the decision back then was to
> not backport it. We're still in beta, so there's some room for arguing
> that we should backport it now, but I'd be happy to leave it until 6.0.
>

I see no reason not to back port it.

Dave Cramer

**Comment by hlinnaka** [comment]
On 08/23/2017 03:03 PM, Dave Cramer wrote:
> On 23 August 2017 at 07:34, Heikki Linnakangas <notifications@github.com>
> wrote:
> 
>> On 08/23/2017 01:49 PM, Dave Cramer wrote:
>>> Is the intent to leave this un-fixed in 5.0?
>>
>> Yeah, I believe so. In PostgreSQL, too, the decision back then was to
>> not backport it. We're still in beta, so there's some room for arguing
>> that we should backport it now, but I'd be happy to leave it until 6.0.
> 
> I see no reason not to back port it.

Well, if you feel inclined, feel free to open a PR. I don't feel 
strongly either way.

- Heikki

**Comment by Eulerizeit** [comment]
@melanieplageman didn't you just fix this?
